### ISSUE TITLE
Using sources confuses `dbt run --models dir+` and runs undesired models

### ISSUE THREAD

**Issue by joshpeng** [fishtown-analytics/dbt#1524]
### Issue description
Prior to switching my repo to use sources I could run `dbt run --models dev+` to only run the exact models inside my `dev` folder and subdirectories. After switching to power all my models with references against sources, when I run the same command, it tries to run models in other directories outside of just `dev`.

Original setup:
1. Models directly called source tables via `var`s defined in `dbt_project.yml`
2. `dbt run --models dir+` worked as expected in running only models in directory/subdir

New setup with dbt sources:
1. Added dbt sources for all of source tables into `schema.yml`
2. Made base tables that called the sources only once
3. Replaced all my old model var references with new `{{ ref('base_source_table') }}` 
4. `dbt run --models dir+` now erroneously runs models outside of the directory listed

Not really sure why or how to debug this because if instead I ran `dbt run --models dev.base+ dev.marketing+` (which is the complete subdirectory structure) everything works correctly again. How come the root `dev` directory doesn't work like prior to using sources?

### Results
Runs more models than what is just in the desired directory.
<img width="386" alt="Screen Shot 2019-06-09 at 10 08 40 AM" src="https://user-images.githubusercontent.com/22549127/59162015-ce2c8200-8a9e-11e9-9343-dcc7888061f1.png">

### System information
The output of `dbt --version`:
```
installed version: 0.13.1
   latest version: 0.13.1

Up to date!
```
OS: `macOS Mojave`
Python: `3.7.3`

**Comment by drewbanin** [comment]
Hey @joshpeng - not sure how i missed this one! This definitely doesn't sound right to me. Were you able to create a minimal reproduction case? Happy to poke around and try to replicate this behavior, just curious if it's something you've already done.

We'll check it out!

**Comment by joshpeng** [comment]
Hi @drewbanin I have not specifically tried to isolate it, but worked around it by changing the front matter config in the models inside the `database_*` folders with this:

```
{{
    config(
        materialized = 'table',
        enabled = (target.type == 'snowflake' and
                    target.name not in ('dev-snowflake', 'dev-snowflake2'))
    )
}}
```
Basically a weird workaround that prevented the models from running from my dev profiles.

**Comment by drewbanin** [comment]
Thanks for the additional info. I tried to reproduce this today but didn't have any luck. If you could provide any other info here, that would be super helpful!

Some questions I have:
1. do any of the models in the `database_*` folders depend on any models/sources defined in the `dev/` folder?
2. do you have any sources named `dev`?
3. is dbt running all of the models in `database_*`, or just some? Are the ones that its running related at all to the models in the `dev` dir?

Thanks!

**Comment by drewbanin** [comment]
closing as i was unable to reproduce. Happy to reopen and fix if anyone can provide a repro case!

**Comment by Anshuman151** [comment]
> Hi @drewbanin I have not specifically tried to isolate it, but worked around it by changing the front matter config in the models inside the `database_*` folders with this:
> 
> ```
> {{
>     config(
>         materialized = 'table',
>         enabled = (target.type == 'snowflake' and
>                     target.name not in ('dev-snowflake', 'dev-snowflake2'))
>     )
> }}
> ```
> 
> Basically a weird workaround that prevented the models from running from my dev profiles.

hi @joshpeng  is there anyway to define it in dbt_project.yml , I am basically developing models as per the adapter requirements in a hierarchical way, for e.g, if i am using postgres adapter , the models that have to be executed were to be present in the "models/postgres/" path
